pkg/runtime/helper_test.go

```diff
@@ -63,6 +63,28 @@ func TestExtractList(t *testing.T) {
 	}
 }
 
+func TestExtractGenericList(t *testing.T) {
```

Comment:
Test the variations of the arrays beyond just `[]interface` and `[]value` - test `[]pointer to interface` and `[]pointer to value`